File: python/developer.py
from employee import Employee
from academic_diploma import AcademicDiploma

from typing import Iterable, List

class Developer(Employee):
    def __init__(self, name: str, diplomas: Iterable[AcademicDiploma]):
        super().__init__(name)
        self._earned: List[AcademicDiploma] = list(diplomas)

    def get_earned(self) -> List[AcademicDiploma]:
        return list(self._earned)

File: java/Developer.java
import java.util.LinkedList;
import java.util.List;

public class Developer extends Employee {
    private List<AcademicDiploma> earned = new LinkedList<>();

    public Developer(String name, Iterable<AcademicDiploma> diplomas) {
        super(name);
        for (AcademicDiploma diploma : diplomas) {
            earned.add(diploma);
        }
    }

    public List<AcademicDiploma> getEarned() { return new LinkedList<>(earned); }
}
